app/controllers/doctors_controller.rb

```diff
@@ -0,0 +1,52 @@
+class DoctorsController < ApplicationController
+  before_action :set_doctor, only: %i[show update destroy]
+
+  # GET /doctors
+  def index
+    @doctors = Doctor.all
+
+    render json: @doctors
+  end
+
+  # GET /doctors/1
+  def show
+    render json: @doctor
+  end
+
+  # POST /doctors
+  def create
+    @doctor = Doctor.new(doctor_params)
+
+    if @doctor.save
+      render json: @doctor, status: :created, location: @doctor
+    else
+      render json: @doctor.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /doctors/1
+  def update
+    if @doctor.update(doctor_params)
+      render json: @doctor
+    else
+      render json: @doctor.errors, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /doctors/1
+  def destroy
+    @doctor.destroy!
+  end
```

Comment:
- [ ] What happens here if the doctor is not successfully destroyed?  Kindly, add some error handling like you did in the #update method (line 28). 👍